js/src/toast.js

```diff
@@ -174,8 +176,53 @@ class Toast extends BaseComponent {
     return config
   }
 
+  _maybeScheduleHide() {
+    if (!this._config.autohide) {
+      return
+    }
+
+    if (this._hasMouseInteraction || this._hasKeyboardInteraction) {
+      return
+    }
+
+    this._timeout = setTimeout(() => {
+      this.hide()
+    }, this._config.delay)
+  }
+
+  _onInteraction(event, isInteracting) {
+    switch (event.type) {
+      case 'mouseover':
+      case 'mouseout':
+        this._hasMouseInteraction = isInteracting
+        break
+      case 'focusin':
+      case 'focusout':
+        this._hasKeyboardInteraction = isInteracting
+        break
+      default:
+        break
+    }
+
+    if (isInteracting) {
+      this._clearTimeout()
+      return
+    }
+
+    const nextElement = event.relatedTarget
+    if (this._element === nextElement || this._element.contains(nextElement)) {
```

Comment:
@RyanBerliner, does #33310  changes the needs for these checks?

Reply:
If we were _only_ concerned with mouse pointer interactions then yes, that event handler PR would allow us to use `mouseenter` and `mouseleave` and omit this check. However, since we are using `focusin` and `focusout` here as well, it's most clear IMO to use the equivalent mouse events and share all the same handler logic. If you have a compelling case otherwise I'm not opposed to changing this.

Reply:
@rohit2sharma95 @alpa any opinions?

Reply:
I'm fine with either way. For mouse events it might be not needed anymore but like Ryan mentioned it makes sense for focus events and it's not like it really bloats the code IMO.